File: src/main/java/editor/gui/editor/CardException.java
package editor.gui.editor;

import java.util.Collection;
import java.util.Collections;

import editor.database.card.Card;

public class CardException extends RuntimeException
{
    public final Collection<Card> cards;

    public CardException(Collection<Card> c, String msg)
    {
        super(c.toString() + (msg.isEmpty() ? "" : ": " + msg));
        cards = Collections.unmodifiableCollection(c);
    }

    public CardException(Card c, String msg)
    {
        this(Collections.singleton(c), msg);
    }

    public CardException(Collection<Card> c)
    {
        this(c, "");
    }

    public CardException(Card c)
    {
        this(c, "");
    }
}

File: src/main/scala/editor/gui/deck/CardException.scala
package editor.gui.deck

import editor.database.card.Card

/**
 * Exception indicating something unexpected went wrong with processing cards.
 * 
 * @constructor create a new exception
 * @param msg string indicating what went wrong
 * @param cards cards involved in the exception
 */
final case class CardException(msg: String = "", cards: Card*)
  extends RuntimeException(s"""${cards.mkString("[", ",", "]")}${if (msg.isEmpty) "" else s": $msg"}""")